core/templates/dev/head/pages/exploration_player/PlayerTranscriptService.js

```diff
@@ -80,27 +85,49 @@ oppia.factory('playerTranscriptService', ['$log', function($log) {
       lastCard.destStateName = newDestStateName;
     },
     addNewAnswer: function(answer) {
-      var pairs = transcript[transcript.length - 1].answerFeedbackPairs;
+      var pairs = transcript[transcript.length - 1].inputResponsePairs;
       if (pairs.length > 0 &&
           pairs[pairs.length - 1].oppiaFeedbackHtml === null) {
         throw Error(
           'Trying to add an answer before the feedback for the previous ' +
           'answer has been received.',
           transcript);
       }
-      transcript[transcript.length - 1].answerFeedbackPairs.push({
+      transcript[transcript.length - 1].inputResponsePairs.push({
         learnerAnswer: answer,
         oppiaFeedbackHtml: null
       });
     },
+    addNewHintRequest: function(hintRequest) {
```

Comment:
Maybe get rid of both this and the answer stuff and reframe as: addNewInput(), addNewResponse() -- since everything here is now reframed as input/response, and that's what you want to see in the transcript anyway.

Only catch is that you'll need to have a lock on the transcript while you're waiting for the response. (How will you prevent the user from submitting a hint requests or an answer while the previous one is being processed, btw? Do you ignore it, prevent them from submitting it, or queue it up?)